pyrightconfig.json

```diff
@@ -2,5 +2,6 @@
   "typeCheckingMode": "strict",
   "reportPrivateUsage": false,
   "reportMissingModuleSource": false,
-  "reportUnnecessaryIsInstance": false
+  "reportUnnecessaryIsInstance": false,
+  "reportIncompatibleMethodOverride": false
```

Comment:
This is because in some places, specially the lookups, some subclasses override the method `as_sql` which receives extra args as just `def as_sql(self):`. This warning is not useful here because the signatures should match the code in django itself